### PR TITLE
IGAPP-912: Fix Linting Warnings

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

Most warnings left now come from array indexing with the non-null-assertiion. 
I was also wondering why we don't have a `node_modules` folder and a `yarn.lock` in the tools folder. Any reason for this?

### CODE REVIEW

**Review by klinzo — CHANGES_REQUESTED** [review]
yarn.lock should be reverted. There is a yarn.lock in the tools dir: https://github.com/digitalfabrik/integreat-app/blob/main/tools/yarn.lock
The node_modules are not synced to git, so the folder is only created once you run `yarn install` in the tools folder

File: yarn.lock [hunk 1]
```diff
@@ -3548,6 +3548,134 @@
     "@nodelib/fs.scandir" "2.1.4"
     fastq "^1.6.0"
 
+"@octokit/auth-app@^2.4.5":
```
[the remark this answers is not in the document]
Comment (Taggotty) [reply yarn.lock:3551]: will revert. not sure what happened here.

**Review by klinzo — APPROVED** [review]

File: native/src/routes/ExternalOfferContainer.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ type PropsType = {
   navigation: NavigationPropType<ExternalOfferRouteType>
 }
 
-const ExternalOfferContainer = ({ route }: PropsType): ReactElement => {
+const ExternalOfferContainer = ({ route, navigation: _navigation }: PropsType): ReactElement => {
```
Comment (SteffiStuffel) [inline native/src/routes/ExternalOfferContainer.tsx:13]: Why not just removing `navigation` from PropsType?

**Review by SteffiStuffel — APPROVED** [review]

File: native/src/routes/ExternalOfferContainer.tsx (hunk 2 above)
Comment (Taggotty) [reply native/src/routes/ExternalOfferContainer.tsx:13]: to keep the information, that navigation is available here, but we do not use it

**Pull request merged**